lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1129,8 +1167,11 @@ def _can_service_start(self) -> bool:
             not self.peers_data.get(Scope.APP, "security_index_initialised", False)
             or not self.alt_hosts
         ):
-            return (
-                self.unit.is_leader() and deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+            return self.unit.is_leader() and (
+                deployment_desc.start == StartMode.WITH_GENERATED_ROLES
+                or deployment_desc.typ == DeploymentType.MAIN_ORCHESTRATOR
+                or deployment_desc.typ == DeploymentType.OTHER
```

Comment:
I believe we can remove this line. A failover orchestrator can also have a "data" role

Reply:
done